### PR TITLE
refactor(db): change diesel to sqlx  

### PR DESCRIPTION
**Pull request by noel2004**
For #6 : 

Now we completely migrate into sqlx, abandon the depending of diesel.

Some hard coding sql statements can be checked by compile-time verification, with the rustc flag "--cfg sqlxverf".

Only batch inserting left for further work.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -44,21 +38,93 @@ fn main() {
...
     let config_file = dotenv::var("CONFIG_FILE")?;
     conf.merge(config_rs::File::with_name(&config_file)).unwrap();
     let settings: config::Settings = conf.try_into().unwrap();
     println!("Settings: {:?}", settings);
 
-    let conn = ConnectionType::establish(&settings.db_log)?;
-    embedded_migrations::run_with_output(&conn, &mut std::io::stdout())?;
-
+    let mut conn = ConnectionType::connect(&settings.db_log).await?;
+    //TODO: add migrations
```
Comment (lispc) [inline src/main.rs:53]: this `TODO` seems outdated?

**Review by lispc — APPROVED** [review]
LGTM. I think this can be squashed and merged once the CI passed

**Comment by lispc** [comment]
CI crashed with `thread 'main' panicked at 'Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.', `

File: migrations/20200123090258_trade_history.sql [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    change        DECIMAL(30,8) NOT NULL,
+    balance       DECIMAL(30,16) NOT NULL,
+    detail        TEXT NOT NULL
+) ;
+
+CREATE INDEX balance_history_idx_user_asset ON balance_history (user_id, asset);
+CREATE INDEX balance_history_idx_user_asset_business ON balance_history (user_id, asset, business);
+
+CREATE TABLE order_history (
+    id            BIGINT CHECK (id >= 0) NOT NULL PRIMARY KEY,
```
Comment (HAOYUatHZ) [inline migrations/20200123090258_trade_history.sql:19]: SERIAL PRIMARY KEY?

File: migrations/20200123090047_trade_log.sql [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+CREATE TABLE slice_history (
+    id            INT CHECK (id >= 0) NOT NULL PRIMARY KEY DEFAULT NEXTVAL ('slice_history_seq'),
+    time          BIGINT NOT NULL,
+    end_operation_log_id   BIGINT CHECK (end_operation_log_id >= 0) NOT NULL,
+    end_order_id  BIGINT CHECK (end_order_id >= 0) NOT NULL,
+    end_trade_id  BIGINT CHECK (end_trade_id >= 0) NOT NULL
+) ;
+
+CREATE TABLE operation_log (
+    id            BIGINT CHECK (id >= 0) NOT NULL PRIMARY KEY,
```
Comment (HAOYUatHZ) [inline migrations/20200123090047_trade_log.sql:45]: SERIAL PRIMARY KEY?

**Comment by HAOYUatHZ** [comment]
没用 Foreign Key Constraint 是否是出于性能考虑

File: migrations/20200123090258_trade_history.sql [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    user_id       INT CHECK (user_id >= 0) NOT NULL,
+    market        VARCHAR(30) NOT NULL,
+    trade_id       BIGINT CHECK (trade_id >= 0) NOT NULL,
+    order_id      BIGINT CHECK (order_id >= 0) NOT NULL,
+    counter_order_id BIGINT CHECK (counter_order_id >= 0) NOT NULL,
+    side          SMALLINT CHECK (side >= 0) NOT NULL,
+    role          SMALLINT CHECK (role >= 0) NOT NULL,
+    price         DECIMAL(30,8) NOT NULL,
+    amount        DECIMAL(30,8) NOT NULL,
+    quote_amount          DECIMAL(30,16) NOT NULL,
```
Comment (HAOYUatHZ) [inline migrations/20200123090258_trade_history.sql:51]: @lispc 原来的为什么  base 精确到 小数点8位 ，quote精确到小数点16位    
比如说币安上有 SNGLS/BTC 交易对
BASE SNGLS 数量有可能很小

Comment (HAOYUatHZ) [reply migrations/20200123090258_trade_history.sql:51]: 不过鉴于这是一个 重构性质的 PR
原来 精度这么配置的话   这个PR 保留也有道理

**Comment by HAOYUatHZ** [comment]
> CI crashed with `thread 'main' panicked at 'Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.', `

Seems current github CI scripts are still not perfect. `26af4f9` this commit passes the tests.

**Review by lispc — APPROVED** [review]

**Pull request merged**